dnsmasq-metrics/pkg/dnsmasq/dnsmasq.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dnsmasq
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/golang/glog"
+	"github.com/miekg/dns"
+)
+
+// MetricsClient is a client used to obtain metrics from dnsmasq.
+type MetricsClient interface {
+	GetMetrics() (ret *Metrics, err error)
+}
+
+type metricsClient struct {
+	addrPort  string
+	dnsClient *dns.Client
+}
+
+const (
+	// CacheHits from dnsmasq.
+	CacheHits = iota
+	// CacheMisses from dnsmasq
+	CacheMisses
+	// CacheEvictions from dnsmasq
+	CacheEvictions
+	// CacheInsertions from dnsmasq
+	CacheInsertions
+	// CacheSize from dnsmasq
+	CacheSize
+	// MetricCount is total number metrics.
+	MetricCount
+)
+
+// MetricToString maps metrics to their string representation.
+var MetricToString = map[int]string{
+	CacheHits:       "hits",
+	CacheMisses:     "misses",
+	CacheEvictions:  "evictions",
+	CacheInsertions: "insertions",
+	CacheSize:       "cachesize",
+}
+
+// Metrics exported by dnsmasq via *.bind CHAOS queries.
+type Metrics [MetricCount]int64
```

Comment:
Optional: fixed-size arrays and *ToString() methods are a very C-ish approach.  If I were doing this, I would a) define `type MetricName string`; b) define the consts as MetricName; c) make this a `map[MetricName]int64` Strings are pretty cheap in Go.